pkg/utils/controller/controller_test.go

```diff
@@ -492,8 +494,13 @@ func TestFetchAllClusterResourceSnapshots(t *testing.T) {
 					return s1 < s2
 				}),
 			}
-			if diff := cmp.Diff(tc.want, got, options...); diff != "" {
-				t.Errorf("FetchAllClusterResourceSnapshots() mismatch (-want, +got):\n%s", diff)
+			theSortedKeys := slices.Sorted(maps.Keys(got))
```

Comment:
Hi Ryan! Just curious: is there anything particular that could stop the direction diff? The key sorting/lookup seems a bit unnecessary?

Reply:
I have tried, the problem is that FetchAllClusterResourceSnapshots returns ResourceSnapshotObj while the got is of type ClusterResourceSnapshot. The cmp lib can't seem to figure that out in a map but it can convert successfully in sclar comparision